packages/core/src/sync-store/postgres/store.ts

```diff
@@ -265,37 +267,34 @@ export class PostgresSyncStore implements SyncStore {
 
     const baseQuery = this.db
       .selectFrom("logs")
-      .select([selectChildAddressExpression.as("childAddress"), "blockNumber"])
+      .select(["id", selectChildAddressExpression.as("childAddress")])
       .where("chainId", "=", chainId)
       .where("address", "=", address)
       .where("topic0", "=", eventSelector)
-      .where("blockNumber", "<=", upToBlockNumber)
+      .where("blockNumber", ">=", fromBlock)
+      .where("blockNumber", "<=", toBlock)
+      .orderBy("id", "asc")
       .limit(pageSize);
 
-    let cursor: bigint | undefined = undefined;
+    let cursor: string | undefined = undefined;
 
     while (true) {
       let query = baseQuery;
-
-      if (cursor) {
-        query = query.where("blockNumber", ">", cursor);
-      }
+      if (cursor !== undefined) query = query.where("id", ">", cursor);
 
       const batch = await this.db.wrap(
         { method: "getFactoryChildAddresses" },
         () => query.execute(),
       );
 
-      const lastRow = batch[batch.length - 1];
-      if (lastRow) {
-        cursor = lastRow.blockNumber;
-      }
-
       if (batch.length > 0) {
         yield batch.map((a) => a.childAddress);
```

Comment:
Just for my understanding, if the page size is 1,000, this will return 1,001 addresses?

Reply:
I don't think so, I think it would still return 1000. The `+ 1` trick we use in `findMany` is required to support the bidirectional pagination, but I don't think we need it here. Maybe you're pointing out something else though - lmk.

Reply:
Makes sense - the pagination logic doesn't attempt to check if the next page is going to have any events or not